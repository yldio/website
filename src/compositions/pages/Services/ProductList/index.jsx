/* eslint-disable global-require */

import React from 'react';

import {
  Container,
  Image,
  Product,
  ProductColumn,
  Title,
  ImageWrapper,
} from './styled';

const ProductList = () => (
  <Container>
    <ProductColumn>
      <ImageWrapper>
        <Image
          src={require('../../../../assets/resource.svg')}
          alt="Strategy"
          width="34px"
          height="40px"
        />
      </ImageWrapper>
      <Title>Strategy</Title>
      <ul>
        <Product>Digital Transformation</Product>
        <Product>Engineering Consultancy</Product>
        <Product>Product Consultancy</Product>
      </ul>
    </ProductColumn>
    <ProductColumn>
      <ImageWrapper>
        <Image
          src={require('../../../../assets/design.svg')}
          alt="Design and Product"
          width="50px"
          height="31px"
        />
      </ImageWrapper>
      <Title>Design &amp; Product</Title>
      <ul>
        <Product>Design Systems &amp; Design Operations</Product>
        <Product>Design Sprints</Product>
        <Product>User Interface Design &amp; Branding</Product>
        <Product>Research, Usability and User Experience</Product>
      </ul>
    </ProductColumn>
    <ProductColumn>
      <ImageWrapper>
        <Image
          src={require('../../../../assets/eng.svg')}
          alt="Engineering"
          width="39px"
          height="32px"
        />
      </ImageWrapper>
      <Title>Engineering</Title>
      <ul>
        <Product>Engineering Leadership</Product>
        <Product>Back End &amp; Node.js</Product>
        <Product>Front End &amp; React.js</Product>
        <Product>DevOps &amp; Kubernetes</Product>
        <Product>Mobile &amp; React Native</Product>
        {/* <Product>Microservices &amp; evolutionary architecture</Product>
        <Product>Cloud infrastructure</Product>
        <Product>Cloud &amp; Containers</Product>
        <Product>Emerging technologies </Product>
        <Product>Machine Learning</Product>
        <Product>.Net Core</Product> */}
      </ul>
    </ProductColumn>
    <ProductColumn>
      <ImageWrapper>
        <Image
          src={require('../../../../assets/training.svg')}
          alt="Engineering"
          width="37px"
          height="32px"
        />
      </ImageWrapper>
      <Title>Training</Title>
      <ul>
        <Product>Node Training</Product>
        <Product>React Training</Product>
        <Product>Design Systems Training</Product>
        <Product>UX Research Training</Product>
        <Product>UX Design Training</Product>
      </ul>
    </ProductColumn>
    <ProductColumn>
      <ImageWrapper>
        <Image
          src={require('../../../../assets/rocket.svg')}
          alt="Addcode"
          width="39px"
          height="39px"
        />
      </ImageWrapper>
      <Title>Addcode</Title>
      <ul>
        <Product>Remote Teams</Product>
        <Product>Geography - same time zone</Product>
        <Product>Scalability</Product>
        <Product>Talent</Product>
        <Product>Choice of location</Product>
        <Product>Lisbon, Porto, Manchester</Product>
      </ul>
    </ProductColumn>
  </Container>
);

export default ProductList;
