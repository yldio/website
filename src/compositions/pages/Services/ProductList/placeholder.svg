<?xml version="1.0" encoding="UTF-8"?>
<svg width="200px" height="128px" viewBox="0 0 200 128" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <defs></defs>
    <rect fill="#4D62D8" width="200" height="128"></rect>
</svg>
