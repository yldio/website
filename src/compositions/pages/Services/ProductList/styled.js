import styled from 'styled-components';
import { spacing } from 'styles/spacing';
import breakpoints from 'styles/breakpoints';
import Panel from 'components/Panel';
import remcalc from 'remcalc';

import { textLight, textLighter } from 'styles/colours';

export const Container = Panel.withComponent('ul').extend`
  display: flex;
  flex-wrap: wrap;
  padding: ${spacing(4)} ${spacing(2)};
  margin-left: ${spacing(-4)};
  margin-right: ${spacing(-4)};
  box-shadow: none;
`;

export const ProductColumn = styled.li`
  font-size: 1.125rem;
  padding: 0 ${spacing(2)} ${spacing(2)} ${spacing(2)};
  position: relative;
  width: 100%;

  ${breakpoints.phablet`
    width: 50%;
  `};

  ${breakpoints.desktop`
    width: 20%;
  `};
`;

export const Image = styled.img`
  height: auto;
`;

export const Product = styled.li`
  color: ${textLighter};
  margin-bottom: ${spacing(2)};
`;

export const Title = styled.header`
  color: ${textLight};
  font-weight: 500;
  margin-bottom: ${spacing(3)};
`;

export const ImageWrapper = styled.div`
  height: ${remcalc(60)};
`;
