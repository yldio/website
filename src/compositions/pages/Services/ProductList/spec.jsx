import React from 'react';
import { shallow } from 'enzyme';
import ProductList from './index';

test('components/ProductList snapshot', () => {
  const wrapper = shallow(<ProductList />);

  expect(wrapper).toMatchSnapshot();
});
