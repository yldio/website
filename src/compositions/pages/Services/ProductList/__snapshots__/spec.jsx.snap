// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ProductList snapshot 1`] = `
<Panel>
  <styled__ProductColumn>
    <styled__ImageWrapper>
      <styled__Image
        alt="Strategy"
        height="40px"
        src="/src/assets/resource.svg"
        width="34px"
      />
    </styled__ImageWrapper>
    <styled__Title>
      Strategy
    </styled__Title>
    <ul>
      <styled__Product>
        Digital Transformation
      </styled__Product>
      <styled__Product>
        Engineering Consultancy
      </styled__Product>
      <styled__Product>
        Product Consultancy
      </styled__Product>
    </ul>
  </styled__ProductColumn>
  <styled__ProductColumn>
    <styled__ImageWrapper>
      <styled__Image
        alt="Design and Product"
        height="31px"
        src="/src/assets/design.svg"
        width="50px"
      />
    </styled__ImageWrapper>
    <styled__Title>
      Design & Product
    </styled__Title>
    <ul>
      <styled__Product>
        Design Systems & Design Operations
      </styled__Product>
      <styled__Product>
        Design Sprints
      </styled__Product>
      <styled__Product>
        User Interface Design & Branding
      </styled__Product>
      <styled__Product>
        Research, Usability and User Experience
      </styled__Product>
    </ul>
  </styled__ProductColumn>
  <styled__ProductColumn>
    <styled__ImageWrapper>
      <styled__Image
        alt="Engineering"
        height="32px"
        src="/src/assets/eng.svg"
        width="39px"
      />
    </styled__ImageWrapper>
    <styled__Title>
      Engineering
    </styled__Title>
    <ul>
      <styled__Product>
        Engineering Leadership
      </styled__Product>
      <styled__Product>
        Back End & Node.js
      </styled__Product>
      <styled__Product>
        Front End & React.js
      </styled__Product>
      <styled__Product>
        DevOps & Kubernetes
      </styled__Product>
      <styled__Product>
        Mobile & React Native
      </styled__Product>
    </ul>
  </styled__ProductColumn>
  <styled__ProductColumn>
    <styled__ImageWrapper>
      <styled__Image
        alt="Engineering"
        height="32px"
        src="/src/assets/training.svg"
        width="37px"
      />
    </styled__ImageWrapper>
    <styled__Title>
      Training
    </styled__Title>
    <ul>
      <styled__Product>
        Node Training
      </styled__Product>
      <styled__Product>
        React Training
      </styled__Product>
      <styled__Product>
        Design Systems Training
      </styled__Product>
      <styled__Product>
        UX Research Training
      </styled__Product>
      <styled__Product>
        UX Design Training
      </styled__Product>
    </ul>
  </styled__ProductColumn>
  <styled__ProductColumn>
    <styled__ImageWrapper>
      <styled__Image
        alt="Addcode"
        height="39px"
        src="/src/assets/rocket.svg"
        width="39px"
      />
    </styled__ImageWrapper>
    <styled__Title>
      Addcode
    </styled__Title>
    <ul>
      <styled__Product>
        Remote Teams
      </styled__Product>
      <styled__Product>
        Geography - same time zone
      </styled__Product>
      <styled__Product>
        Scalability
      </styled__Product>
      <styled__Product>
        Talent
      </styled__Product>
      <styled__Product>
        Choice of location
      </styled__Product>
      <styled__Product>
        Lisbon, Porto, Manchester
      </styled__Product>
    </ul>
  </styled__ProductColumn>
</Panel>
`;
